conanfile.py

```diff
@@ -266,34 +280,41 @@ def _openssl_path(self):
             pck = "OpenSSL"
         return self.conanfile.deps_cpp_info[pck].rootpath
 
-    def _build(self, dest_dir):
-        from conan.tools.gnu import AutotoolsToolchain, Autotools
-
+    def generate(self, dest_dir):
+        from conan.tools.gnu import AutotoolsToolchain, AutotoolsDeps
         tc = AutotoolsToolchain(self.conanfile, prefix=dest_dir)
         tc.configure_args.extend(["--enable-shared", f"--with-openssl={self._openssl_path}"])
-        tc.make_args.append("-j8")
+        tc.generate()
 
+        tc = AutotoolsDeps(self.conanfile)
         # We need to do this manually because `AutotoolsDeps` doesn't add `tools_requires` deps
         for dep in self.conanfile.deps_cpp_info.deps:
             info = self.conanfile.deps_cpp_info[dep]
-            tc.extra_cflags.extend([f"-I{x}" for x in info.include_paths])
-            tc.extra_ldflags.extend([f"-L{x}" for x in info.lib_paths])
+            for x in info.include_paths:
+                tc.environment.append("CPPFLAGS", f"-I{x}")
+            for x in info.lib_paths:
+                tc.environment.append("LDFLAGS", f"-L{x}")
 
         # On Linux, we need to set RPATH so that `root/bin/python3` can correctly find the `.so`
         # file in `root/lib` no matter where `root` is. We need it to be portable. We explicitly
         # set `--disable-new-dtags` to use RPATH instead of RUNPATH. RUNPATH can be overridden by
         # the LD_LIBRARY_PATH env variable which is not at all what we want for this self-contained
         # package. Unlike RUNPATH, RPATH takes precedence over LD_LIBRARY_PATH.
         if self.conanfile.settings.os == "Linux":
-            tc.extra_ldflags.extend(["-Wl,-rpath,'$$ORIGIN/../lib'", "-Wl,--disable-new-dtags"])
+            for x in ["-Wl,-rpath='\$\$ORIGIN/../lib'", "-Wl,--disable-new-dtags"]:
+                tc.environment.append("LDFLAGS", x)
 
         tc.generate()
 
-        with tc.environment().vars(self.conanfile).apply():
-            autotools = Autotools(self.conanfile)
-            autotools.configure()
-            autotools.make()
-            autotools.install(args=["DESTDIR=''"])  # already handled by `prefix=dest_dir`
+    def build(self, prefix_dir):
+        autotools = Autotools(self.conanfile)
+        autotools.configure()
+        autotools.make()
+
+    def install(self, dest_dir):
+        autotools = Autotools(self.conanfile)
+        autotools.install(
+            args=["DESTDIR=''"])  # already handled by `prefix=dest_dir`autotools = Autotools(self.conanfile)
```

Comment:
This comment got mangled with some code it seems:
```suggestion
        autotools.install(args=["DESTDIR=''"])  # already handled by `prefix=dest_dir`
```